core/src/main/java/hudson/triggers/SCMTrigger.java

```diff
@@ -313,7 +319,7 @@ public void setPollingThreadCount(int n) {
 
         @Restricted(NoExternalUse.class)
         public boolean isPollingThreadCountOptionVisible() {
-            if (getPollingThreadCount() != 0) {
+            if (getPollingThreadCount() != 10) {
                 // this is a user who already configured the option
```

Comment:
Why does this logic even exist again?

Reply:
WDYM? Not showing an advanced option unless it's likely to be needed?

Reply:
Yeah I am not sure why code like this needs to exist. The normal UI is an **Advanced…** button, and if the value has been edited from its default (which can be inferred automatically by normal databinding), an “edited” icon appears on it.

Reply:
There's no non-advanced UI though, so it would be just a section title and an Advanced button?

Reply:
Sure.

To be clear, I am not trying to block this, just noting a related UI simplification we could make.